pkg/local_object_storage/metabase/metadata.go

```diff
@@ -0,0 +1,796 @@
+}
+
+// TODO: fill on migration.
+// TODO: ROOT and PHY props.
+// TODO: cleaning on obj removal.
+func putMetadata(tx *bbolt.Tx, cnr cid.ID, id oid.ID, ver version.Version, owner user.ID, typ object.Type, creationEpoch uint64,
+	payloadLen uint64, pldHash, pldHmmHash, splitID []byte, parentID, firstID oid.ID, attrs []object.Attribute) error {
+	metaBkt, err := tx.CreateBucketIfNotExists(metaBucketKey(cnr))
+	if err != nil {
+		return fmt.Errorf("create meta bucket for container: %w", err)
+	}
+	idk := [1 + oid.Size]byte{metaPrefixID}
+	copy(idk[1:], id[:])
+	if err := metaBkt.Put(idk[:], nil); err != nil {
+		return fmt.Errorf("put object ID to container's meta bucket: %w", err)
+	}
+
+	var keyBuf keyBuffer
+	if err = putPlainAttribute(metaBkt, &keyBuf, id, object.FilterVersion, ver.String()); err != nil {
+		return err
+	}
+	if err = putPlainAttribute(metaBkt, &keyBuf, id, object.FilterOwnerID, string(owner[:])); err != nil {
+		return err
+	}
+	if err = putPlainAttribute(metaBkt, &keyBuf, id, object.FilterType, typ.String()); err != nil {
+		return err
+	}
+	if err = putIntAttribute(metaBkt, &keyBuf, id, object.FilterCreationEpoch, new(big.Int).SetUint64(creationEpoch)); err != nil {
+		return err
+	}
+	if err = putIntAttribute(metaBkt, &keyBuf, id, object.FilterPayloadSize, new(big.Int).SetUint64(payloadLen)); err != nil {
+		return err
+	}
+	if err = putPlainAttribute(metaBkt, &keyBuf, id, object.FilterPayloadChecksum, string(pldHash)); err != nil {
+		return err
+	}
+	if err = putPlainAttribute(metaBkt, &keyBuf, id, object.FilterPayloadHomomorphicHash, string(pldHmmHash)); err != nil {
+		return err
+	}
+	if len(splitID) > 0 {
+		if err = putPlainAttribute(metaBkt, &keyBuf, id, object.FilterSplitID, string(splitID)); err != nil {
+			return err
+		}
+	}
+	if !firstID.IsZero() {
+		if err = putPlainAttribute(metaBkt, &keyBuf, id, object.FilterFirstSplitObject, string(firstID[:])); err != nil {
+			return err
+		}
+	}
+	if !parentID.IsZero() {
+		if err = putPlainAttribute(metaBkt, &keyBuf, id, object.FilterParentID, string(parentID[:])); err != nil {
+			return err
+		}
+	}
+	for i := range attrs {
+		ak, av := attrs[i].Key(), attrs[i].Value()
+		if n, isInt := parseInt(av); isInt && n.Cmp(maxUint256Neg) >= 0 && n.Cmp(maxUint256) <= 0 {
+			err = putIntAttribute(metaBkt, &keyBuf, id, ak, n)
+		} else {
+			err = putPlainAttribute(metaBkt, &keyBuf, id, ak, av)
+		}
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// SearchCursor is a cursor used for continuous search in the DB.
+type SearchCursor struct {
+	Key      []byte // PREFIX_ATTR_DELIM_VAL_ID, prefix is unset
+	ValIDOff int
+}
+
+// NewSearchCursorFromString decodes cursor from the string according to the
+// primary attribute.
+func NewSearchCursorFromString(s, primAttr string) (*SearchCursor, error) {
+	if s == "" {
+		return nil, nil
+	}
+	b := make([]byte, 1+base64.StdEncoding.DecodedLen(len(s)))
```

Comment:
isnt `DecodeString` about the same but shorter?

Reply:
`DecodeString` creates new buffer on his own, this isnt desired here